src/core/server/core_usage_data/core_usage_data_service.ts

```diff
@@ -256,6 +262,87 @@ export class CoreUsageDataService implements CoreService<CoreUsageDataSetup, Cor
     };
   }
 
+  private getMarkedAsSafe(
+    exposedConfigsToUsage: ExposedConfigsToUsage,
+    pluginId?: string,
+    usedPath?: string
+  ): { explicitlyMarked: boolean; isSafe: boolean } {
+    if (!pluginId || !usedPath) return { explicitlyMarked: false, isSafe: false };
+    const exposeDetails = exposedConfigsToUsage.get(pluginId);
+    if (!exposeDetails) {
+      return { explicitlyMarked: false, isSafe: false };
+    }
+
+    const exposeKeyDetails = Object.keys(exposeDetails).find((exposeKey) => {
+      const fullPath = `${pluginId}.${exposeKey}`;
+      const hasIntersection = hasConfigPathIntersection(usedPath, fullPath);
+
+      return hasIntersection;
+    });
+    if (!exposeKeyDetails) {
+      return { explicitlyMarked: false, isSafe: false };
+    }
+
+    const isSafe = exposeDetails[exposeKeyDetails];
+    if (typeof isSafe === 'boolean') {
+      return { explicitlyMarked: true, isSafe };
+    }
+
+    return { explicitlyMarked: false, isSafe: false };
```

Comment:
Super-NIT: I wonder if it'd be easier to read if we go through the positive path (`explicitlyMarked: true`) and return `{false, false}` otherwise.

```suggestion
    if (pluginId && usedPath) {
      const exposeDetails = exposedConfigsToUsage.get(pluginId) || {};
      const exposeKeyDetails = Object.keys(exposeDetails).find((exposeKey) => {
        const fullPath = `${pluginId}.${exposeKey}`;
        const hasIntersection = hasConfigPathIntersection(usedPath, fullPath);

        return hasIntersection;
      });
      if (exposeKeyDetails) {
        const isSafe = exposeDetails[exposeKeyDetails];
        if (typeof isSafe === 'boolean') {
          return { explicitlyMarked: true, isSafe };
        }
      }
    }
    
    return { explicitlyMarked: false, isSafe: false };
```